src/Core/Grid/Query/DoctrineSearchCriteriaApplicator.php

```diff
@@ -0,0 +1,67 @@
+<?php
+/**
+ * 2007-2018 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Grid\Query;
+
+use Doctrine\DBAL\Query\QueryBuilder;
+use PrestaShop\PrestaShop\Core\Grid\Search\SearchCriteriaInterface;
+
+/**
+ * Class DoctrineSearchCriteriaApplicator applies search criteria to doctrine query builder
+ */
+final class DoctrineSearchCriteriaApplicator implements DoctrineSearchCriteriaApplicatorInterface
```

Comment:
Not sure if it's really necessary to use the word "Doctrine" everywhere

Reply:
We try to keep the opportunity to remove Doctrine at some point: this probably won't happens, but this make the grid component not hardly bounded to the ORM :+1:

Reply:
@eternoendless i wanted to make clear separation as we could have different applicators for search criteria. For example, backups list data is loaded from filesystem so we dont touch doctrine there at all.